xmodule/capa/safe_exec/tests/test_safe_exec.py

```diff
@@ -370,32 +370,63 @@ def remote_exec(data):
+            '\\n    ^\\nSyntaxError: invalid syntax\\n\' with status code: 1'
+        )
+        side_2 = (
+            'Couldn\'t execute jailed code: stdout: b\'\', stderr: b\'Traceback'
+            ' (most recent call last):\\n  File "jailed_code"'
+            ', line 19, in <module>\\n    exec(code, g_dict)\\n  File "<string>"'
+            ', line 203, in <module>\\n  File "<string>", line 89, in test_add\\n'
+            '  File "<string>", line 1\\n    import random random.choice(range(10))'
+            '\\n    ^^^^^^\\nSyntaxError: invalid syntax\\n\' with status code: 1'
+        )
+        assert normalize_error_message(side_1) == normalize_error_message(side_2)
+
     @override_settings(CODEJAIL_DARKLAUNCH_EMSG_NORMALIZERS=[
-        {
-            'search': r'/tmp/codejail-[0-9a-zA-Z]+',
-            'replace': r'/tmp/codejail-<RAND>',
-        },
         {
             'search': r'[0-9]+',
             'replace': r'<NUM>',
         },
     ])
     def test_configurable_normalizers(self):
-        """We can override the normalizers, and they run in order."""
+        """We can augment the normalizers, and they run in order."""
+        emsg_in = "Error in /tmp/codejail-1234abcd/whatever.py: something 12 34 other"
+        expect_out = "Error in /tmp/codejail-<SANDBOX_DIR_NAME>/whatever.py: something <NUM> <NUM> other"
+        assert expect_out == normalize_error_message(emsg_in)
+
+    @override_settings(
+        CODEJAIL_DARKLAUNCH_EMSG_NORMALIZERS=[
+            {
+                'search': r'[0-9]+',
+                'replace': r'<NUM>',
+            },
+        ],
+        CODEJAIL_DARKLAUNCH_EMSG_NORMALIZERS_COMBINE='replace',
+    )
+    def test_can_replace_normalizers(self):
+        """We can replace the normalizers."""
         emsg_in = "Error in /tmp/codejail-1234abcd/whatever.py: something 12 34 other"
-        expect_out = "Error in /tmp/codejail-<RAND>/whatever.py: something <NUM> <NUM> other"
+        expect_out = "Error in /tmp/codejail-<NUM>abcd/whatever.py: something <NUM> <NUM> other"
         assert expect_out == normalize_error_message(emsg_in)
 
     @override_settings(CODEJAIL_DARKLAUNCH_EMSG_NORMALIZERS=[
         {
-            'search': r'broken [',
-            'replace': r'replace',
+            'search': r'broken',
+            'replace': r'replace \g<>',  # invalid replacement pattern
         },
     ])
     @patch('xmodule.capa.safe_exec.safe_exec.record_exception')
     def test_normalizers_validate(self, mock_record_exception):
-        """Normalizers are validated, and fall back to empty list on error."""
-        assert emsg_normalizers() == []  # pylint: disable=use-implicit-booleaness-not-comparison
+        """Normalizers are validated, and fall back to default list on error."""
+        assert len(emsg_normalizers()) > 0  # pylint: disable=use-implicit-booleaness-not-comparison
         mock_record_exception.assert_called_once()
```

Comment:
Not a big deal, but you could check that it logs as well.

Reply:
Good point, pushed up a log check as well.